tests/compile-fail/while_loop.rs

```diff
@@ -53,6 +53,33 @@ fn main() {
     while let Some(x) = y { // no error, obviously
         println!("{}", x);
     }
+
+    let mut iter = 1..20;
+    while let Option::Some(x) = iter.next() { //~ERROR this loop could be written as a `for` loop
+        println!("{}", x);
+    }
+
+    let mut iter = 1..20;
+    while let Some(x) = iter.next() { //~ERROR this loop could be written as a `for` loop
+        println!("{}", x);
+    }
```

Comment:
No, for loops consume the iterator (probably because `IntoIterator::into_iter` takes `self` by value). I actually considered that, but couldn't come up with convincing use-cases where you would want to use the iterator after the loop. On the other hand, it shouldn't be very hard to fix.